### ISSUE TITLE
num_workers in Dataloader

### ISSUE THREAD

**Issue by MaKaili** [rusty1s/pytorch_geometric#2473]
From now on, we recommend using our discussion forum (https://github.com/rusty1s/pytorch_geometric/discussions) for general questions.

## ❓ Questions & Help

I am trying to use multi-process data loading for a collection of graphs with small nodes number.
`train_loader = DataLoader(dataset, batch_size=batch_size, shuffle=True, num_workers=8)`
The number of running CPU cores is supposed to be equal to num_workers while the data loader processing from my understanding. However, when I trace the CPU utilization, I found it was 100% rather than 800%.
I tried to set OMP_NUM_THREADS=8 when I run the program, but it didn't work.
My question is: when the num_workers=8, the CPU utilization is supposed to be 800%, is it right? If it is true, are there any hints for the right multi-process data loading way? I really appreciate any help you can provide.

**Comment by rusty1s** [comment]
Yes, 800% should be ideally. We directly make use of PyTorch's `DataLoader` and `num_workers` capabilities. Do you see such low CPU utilization in other non-PyG datasets as well? Which OS/PyTorch version are you running on?

**Comment by MaKaili** [comment]
Thanks for the reply. I think I found the problem. I simply pass a regular python list holding `torch_geometric.data.Data` objects and pass them to `torch_geometric.data.DataLoader`. We could batch the data in this way, but it cannot support multi-processing data loading since it doesn't have `collate_fn`.
